core/indigo-core/reaction/src/reaction_cdxml_saver.cpp

```diff
@@ -387,8 +387,22 @@ void ReactionCdxmlSaver::_addStep(BaseReaction& rxn, MoleculeCdxmlSaver& molsave
 void ReactionCdxmlSaver::_generateCdxmlObjIds(BaseReaction& rxn, std::unordered_map<int, int>& mol_ids, std::unordered_map<int, int>& meta_ids,
                                               ObjArray<Array<int>>& nodes_ids, int& arrow_id)
 {
+    int id = 1;
+    arrow_id = id++;
+
+    // generate ids for meta objects. 1 node and 1 extra object. text or graphics
+    for (auto i = 0; i < rxn.meta().metaData().size(); ++i)
+    {
+        int r_id = i + rxn.end();
+        id++;
+        meta_ids.insert(std::make_pair(r_id, id));
+        nodes_ids.expand(r_id + 1);
+        nodes_ids[r_id].clear_resize(1);
+        nodes_ids[r_id].zerofill();
+        id += 2;
```

Comment:
why id++ above and id+=2?

Reply:
There is a comment above:     // generate ids for meta objects. 1 node and 1 extra object. text or graphics
Every metaobject requires two ids.  First is a meta object id itself and the second one for nested text or graphics element so we can easily guess the nested block id from meta object id.